toponetx/classes/combinatorial_complex.py

```diff
@@ -447,9 +448,256 @@ def add_cell(self, cell, rank=None, **attr):
+            )  # incidence is defined between two skeletons of different ranks
+            if (
+                rank < to_rank
+            ):  # up incidence is defined between two skeletons of different ranks
+                children = self.skeleton(rank)
+                uidset = self.skeleton(to_rank)
+
+            elif (
+                rank > to_rank
+            ):  # up incidence is defined between two skeletons of different ranks
+                children = self.skeleton(to_rank)
+                uidset = self.skeleton(rank)
+        return compute_set_incidence(children, uidset, sparse, index)
+
+    def incidence_matrix(
+        self,
+        rank,
+        to_rank=None,
+        incidence_type: str = "up",
+        weight=None,
+        sparse: bool = True,
+        index: bool = False,
+    ):
+        """Compute incidence matrix for the CC between rank and to_rank skeleti.
+
+        Parameters
+        ----------
+        weight : bool, default=False
+            If False all nonzero entries are 1.
+            If True and self.static all nonzero entries are filled by
+            self.cells.cell_weight dictionary values.
+        index : boolean, optional, default False
+            If True return will include a dictionary of node uid : row number
+            and cell uid : column number
+
+        Returns
+        -------
+        incidence_matrix : scipy.sparse.csr.csr_matrix or np.ndarray
+        row dictionary : dict
+            Dictionary identifying rows with nodes
+        column dictionary : dict
+            Dictionary identifying columns with cells
+        """
+        return self._incidence_matrix(
+            rank, to_rank, incidence_type=incidence_type, sparse=sparse, index=index
+        )
+
+    def adjacency_matrix(self, rank, via_rank, s=1, index=False):
+        """Sparse weighted :term:`s-adjacency matrix`.
+
+        Parameters
+        ----------
+        r,k : int, int
+            Two ranks for skeletons in the input Colored Hypergraph
+        s : int, list, optional, default : 1
+            Minimum number of edges shared by neighbors with node.
+        index: boolean, optional, default: False
+            If True, will return a rowdict of row to node uid
+        index : book, default=False
+            indicate weather to return the indices of the adjacency matrix.
+
+        Returns
+        -------
+        If index is True
+            adjacency_matrix : scipy.sparse.csr.csr_matrix
+            row dictionary : dict
+
+        If index if False
+            adjacency_matrix : scipy.sparse.csr.csr_matrix
+
+        Examples
+        --------
+        >>> G = Graph() # networkx graph
+        >>> G.add_edge(0, 1)
+        >>> G.add_edge(0,3)
+        >>> G.add_edge(0,4)
+        >>> G.add_edge(1, 4)
+        >>> CHG = ColoredHyperGraph(cells=G)
+        >>> CHG.adjacency_matrix(0, 1)
+        """
+        if via_rank is not None:
+            assert rank < via_rank
```

Comment:
Don't use `assert` for this. Raise a proper `ValueError` with meaningful error message please.